Source/Core/DolphinQt/Config/VerifyWidget.cpp

```diff
@@ -0,0 +1,145 @@
+// Copyright 2019 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#include "DolphinQt/Config/VerifyWidget.h"
+
+#include <memory>
+#include <vector>
+
+#include <QByteArray>
+#include <QHBoxLayout>
+#include <QHeaderView>
+#include <QLabel>
+#include <QProgressDialog>
+#include <QPushButton>
+#include <QVBoxLayout>
+
+#include "Common/CommonTypes.h"
+#include "DiscIO/Volume.h"
+#include "DiscIO/VolumeVerifier.h"
+
+VerifyWidget::VerifyWidget(std::shared_ptr<DiscIO::Volume> volume) : m_volume(std::move(volume))
```

Comment:
We usually keep the widget creation out of the constructor and split it into two or three methods:
1. ``CreateWidgets()``
2. ``CreateLayout()`` (I'd omit that one here)
3. ``ConnectWidgets()``

Reply:
So I should move all the stuff that is above the line `layout->addWidget(...` to `CreateWidgets()`, except for the `connect` line, which I put in `ConnectWidgets()`?

Reply:
Yep.